flink/v1.15/flink/src/test/java/org/apache/iceberg/flink/sink/TestIcebergStreamWriter.java

```diff
@@ -233,7 +233,31 @@ public void testBoundedStreamCloseWithEmittingDataFiles() throws Exception {
       Assert.assertEquals(expectedDataFiles * 2, result.dataFiles().length);
     }
   }
+  @Test
+  public void testBoundedStreamTriggeredEndInputBeforeTriggeringCheckpoint() throws Exception {
```

Comment:
@pvary can you help trigger the CI build for this PR. I would assume the above test should fail.

Reply:
> thx for adding the unit test. I am a little concerned with one of the existing unit test. why do we want to make sure if `endInput` is called twice, the writer is emitting the result twice. This behavior is a little odd to me. maybe we should dig out the history behind this test.
> 
> ```
>   @Test
>   public void testBoundedStreamCloseWithEmittingDataFiles() throws Exception {
>     try (OneInputStreamOperatorTestHarness<RowData, WriteResult> testHarness =
>         createIcebergStreamWriter()) {
>       testHarness.processElement(SimpleDataUtil.createRowData(1, "hello"), 1);
>       testHarness.processElement(SimpleDataUtil.createRowData(2, "world"), 2);
> 
>       Assertions.assertThat(testHarness.getOneInputOperator()).isInstanceOf(BoundedOneInput.class);
>       ((BoundedOneInput) testHarness.getOneInputOperator()).endInput();
> 
>       long expectedDataFiles = partitioned ? 2 : 1;
>       WriteResult result = WriteResult.builder().addAll(testHarness.extractOutputValues()).build();
>       Assert.assertEquals(0, result.deleteFiles().length);
>       Assert.assertEquals(expectedDataFiles, result.dataFiles().length);
> 
>       // invoke endInput again.
>       ((BoundedOneInput) testHarness.getOneInputOperator()).endInput();
> 
>       result = WriteResult.builder().addAll(testHarness.extractOutputValues()).build();
>       Assert.assertEquals(0, result.deleteFiles().length);
>       Assert.assertEquals(expectedDataFiles * 2, result.dataFiles().length);
>     }
>   }
> ```

Yes, the test failed, should we delete it, or should we modify it

Reply:
The current test will fail because of the write change

Reply:
I think there is something wrong with this test, we need to adjust it, calling EndInput again should not send dataflies again

Reply:
@xuzhiwen1255 I also think it is a little weird that calling endInput twice would send data files again. It is there since the very beginning of this test class. https://github.com/apache/iceberg/commit/7101c25b91287578e387f8c8cdc7f5547310171e#diff-7f9a59307eed16d49e8590463e52c94e2a2e5c46dd132b635991ffa7571ee164

cc @openinx in case there is some context that we missed.

Reply:
I think this test is meaningless. Should we delete it?

Reply:
we should keep the test as it verifies the `endInput` behavior. your adjustment looks good to me

Reply:
Thank you.